### PR TITLE
First version of the installer

### PR DESCRIPTION
**Pull request by lhstrh**
- [x] detect platform and customize accordingly (see [SO](https://stackoverflow.com/questions/394230/how-to-detect-the-os-from-a-bash-script))
- [x] confirm it works on Linux
- [x] confirm it works on Mac
- [x] confirm it works on Windows (WSL)
- [x] add CI
- [x] handle cli args properly

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# Installer for Lingua Franca CLI tools.
+# 
+# Usage: curl -s https://lf-lang.org/install-cli.sh | bash -s nightly
+
+if [[ $1 == "nightly" ]]; then
+  kind="nightly"
+  rel="https://api.github.com/repos/lf-lang/lingua-franca/releases/tags/nightly"
```
Comment (cmnrd) [inline install.sh]: It seems like a bad idea to download stuff from the internet and then install it unchecked. There should be at least a checksum validation step. This, however, would require to also release the install script nightly with the checksums updated.

Comment (lhstrh) [reply install.sh]: Isn't this covered by the fact that we use HTTPS already?

**Review by edwardalee — CHANGES_REQUESTED** [review]
Looks good, except that .bashrc is not universal.  My system uses .bash_profile, for example. I think there are several other alternatives, sadly.

**Comment by lhstrh** [comment]
> Looks good, except that .bashrc is not universal. My system uses .bash_profile, for example. I think there are several other alternatives, sadly.

Yes. This script, if we decide to rely on it, will require a ton of platform-dependent customization...

File: install.sh (hunk 1 above)
Comment (cmnrd) [reply install.sh]: No, because we don't validate the certificate. Anyone could redirect this request by manipulating the DNS for instance.

Comment (lhstrh) [reply install.sh]: Then let's check the certificate?

**Comment by cmnrd** [comment]
Manipulating the .bashrc seems like a bad idea to me. For one, because it only works if the user relies on bash. But more importantly: an installer shouldn't mess with the user's configuration files. There are many Linux users (including me) who are pedantic about this and wouldn't want to come near a software that messes with their precious configuration.

**Comment by lhstrh** [comment]
> Manipulating the .bashrc seems like a bad idea to me. For one, because it only works if the user relies on bash. But more importantly: an installer shouldn't mess with the user's configuration files. There are many Linux users (including me) who are pedantic about this and wouldn't want to come near a software that messes with their precious configuration.

Jep. I agree. This was just a first stab to see whether I could make it work locally. Modifying `.bashrc` is not something we will be doing with the eventual script. I'll mark this PR ready for review when it's ready :rofl:

File: install.sh (hunk 1 above)
Comment (cmnrd) [reply install.sh]: I realized that I was wrong about that. Https protects against DNS spoofing. If we trust the github API to deliver unmanipulated artifacts and all LF maintainers, then checksums don't provide additional protection.

**Pull request merged**